File: backend/internal/dto/sonar_run.go
package dto

import (
	"encoding/json"
	"time"
)

type ImportSonarRunRequest struct {
	TransectPlanID    uint            `json:"transect_plan_id" binding:"required,gt=0"`
	RunCode           string          `json:"run_code" binding:"required,min=3,max=48"`
	TrackGeoJSON      json.RawMessage `json:"track_geojson" binding:"required"`
	ActualSwathM      float64         `json:"actual_swath_m" binding:"required,gt=0,lte=2000"`
	StartedAt         time.Time       `json:"started_at" binding:"required"`
	EndedAt           time.Time       `json:"ended_at" binding:"required"`
	NavigationQuality string          `json:"navigation_quality" binding:"required,oneof=good degraded invalid"`
}

type RunTransitionRequest struct {
	TargetState     string `json:"target_state" binding:"required,oneof=quality_checked processing processed rejected superseded"`
	ExpectedVersion uint   `json:"expected_version" binding:"required,gt=0"`
	Reason          string `json:"reason" binding:"omitempty,max=500"`
}

type SonarRunQuery struct {
	TransectPlanID uint
	State          string
	Quality        string
	Page           int
	PageSize       int
}

type RunQualityEvidence struct {
	CoordinateCount int      `json:"coordinate_count"`
	TrackLengthM    float64  `json:"track_length_m"`
	DurationMinutes float64  `json:"duration_minutes"`
	AverageSpeedMPS float64  `json:"average_speed_mps"`
	Warnings        []string `json:"warnings"`
}

File: frontend/src/types/sonar-run.ts
import type { GeoJSONFeature } from './api'
import type { TransectPlan } from './transect-plan'
import type { RunState } from './enums/run-state'

export interface SonarRun {
  id:number; transect_plan_id:number; run_code:string; track_geojson:GeoJSONFeature; actual_swath_m:number; started_at:string; ended_at:string
  navigation_quality:'good'|'degraded'|'invalid'; run_state:RunState; source_checksum:string; imported_by:number; version:number; created_at:string; updated_at:string
  transect_plan?:TransectPlan
}
export interface ImportSonarRun { transect_plan_id:number; run_code:string; track_geojson:GeoJSONFeature; actual_swath_m:number; started_at:string; ended_at:string; navigation_quality:'good'|'degraded'|'invalid' }
export interface RunQualityEvidence { coordinate_count:number; track_length_m:number; duration_minutes:number; average_speed_mps:number; warnings:string[] }
